### DISCUSSION TITLE
Add support for DNS-01 challenge

### DISCUSSION THREAD

**Discussion by wswv** [nextcloud/all-in-one#493] [Ideas] [answered]
I installed the docker and found it could not be used due to it only access 443 port for domain, bu for most of homer user, ISP blocked the port 443 and 80. some SSL certification could be certified by HTTP-01 challenge. so could add the function(DNS-01) for it? thanks!

**Comment by szaimen** [comment]
This is a new feature so moving to discussions.

**Comment by szaimen** [comment]
Honestly I don't think this is worth being implemented as you can already configure the dns challenge using a reverse proxy. 

Here is a general documentation on using it behind a reverse proxy: https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md

And here is how to configure caddy (as your reverse proxy) to use the dns challenge: https://caddy.community/t/how-to-use-dns-provider-modules-in-caddy-2/8148

**Comment by wswv** [comment]
> This is a new feature so moving to discussions.

OK, Hope the new feature could solve it, or the reverse proxy could have optional solution, so we can use self-defined proxy to solve it. otherwise it can not go to next step fully. thanks!

**Reply by szaimen** [reply]
yes, you can use a reverse proxy to do that just like explained or doesn't it work? What is the error message that you get?

**Comment by szaimen** [comment, marked as answer]
I am adding some documentation for this with https://github.com/nextcloud/all-in-one/issues/876
